lua/lean/_util.lua

```diff
@@ -213,11 +213,33 @@ function M.position_params_valid(params)
 
   if line > #lines then return false end
 
-  if col > vim.fn.strwidth(lines[line]) then return false end
+  if col > #lines[line] then return false end
 
   return true
 end
 
+-- REPLACEME: with vim.lsp.util.make_position_params(nil, "utf-8") after neovim/neovim#16382 is merged and
+-- we drop support for <0.6.0
+local function make_utf8_params()
+  local buf = vim.api.nvim_win_get_buf(0)
+  local row, col = unpack(vim.api.nvim_win_get_cursor(0))
+  row = row - 1
+  local line = vim.api.nvim_buf_get_lines(buf, row, row+1, true)[1]
```

Comment:
The `vim.api.nvim_buf_line_count` function should be enough here, and doesn't require transfer of any buffer contents. (if it's necessary to check that the line is in range at all).

Reply:
Yea sorry this was dead code, removed.